src/main/java/com/linecorp/armeria/client/routing/ServerGroup.java

```diff
@@ -0,0 +1,27 @@
+/*
+ * Copyright 2016 LINE Corporation
+ *
+ * LINE Corporation licenses this file to you under the Apache License,
+ * version 2.0 (the "License"); you may not use this file except in compliance
+ * with the License. You may obtain a copy of the License at:
+ *
+ *   http://www.apache.org/licenses/LICENSE-2.0
+ *
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS, WITHOUT
+ * WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied. See the
+ * License for the specific language governing permissions and limitations
+ * under the License.
+ */
+package com.linecorp.armeria.client.routing;
+
+import java.util.List;
+
+public interface ServerGroup<T extends Node> {
+    List<T> nodes();
+
+    default boolean isReplaceable() {
```

Comment:
I see. If you are worried about mistakenly registering twice, how about just:

- make `register()` fail when a group with the same name is given
- add `replace()` operation that is dedicated to replacement, which succeeds only when a group with the same name exists.

This way, we could probably remove the 'replaceable' property completely? WDYT?